src/app/lib/__tests__/preventChangeOnNumberInputScroll.test.tsx

```diff
@@ -0,0 +1,21 @@
+import { render, fireEvent, act, waitFor } from '@testing-library/react'
+import { TextInput } from 'grommet/es6/components/TextInput'
+import { preventChangeOnNumberInputScroll } from '../preventChangeOnNumberInputScroll'
+
+it('preventChangeOnNumberInputScroll', async () => {
+  const { getByRole } = render(
+    <TextInput role="input" type="number" {...preventChangeOnNumberInputScroll()} />,
+  )
+
+  const numberInput = getByRole('input')
+
+  numberInput.blur = jest.fn()
+  const focusMock = jest.spyOn(numberInput, 'focus')
+
+  act(() => {
+    fireEvent.wheel(numberInput)
+  })
+
+  expect(numberInput.blur).toHaveBeenCalled()
+  await waitFor(() => expect(focusMock).toHaveBeenCalled())
+})
```

Comment:
This tests implementation details instead of the behavior

Reply:
Right, the right implementation would be:
- set value to x
- trigger wheel event scroll
- check if value equal to x

But the issue is that jest probably mocks wheel event, at least I couldn't make it work. So I settled on this - which would check if the implementation calls went through for the on wheel event.

Reply:
here's an e2e test in playwright that fails on master:
```ts
  const input = page.getByPlaceholder('Enter an amount')
  await input.click()
  await input.fill('1111')
  const inputPosition = await input.boundingBox()
  await page.mouse.move(
    inputPosition!.x + inputPosition!.width / 2,
    inputPosition!.y + inputPosition!.height / 2,
  )
  await page.mouse.wheel(0, 10)
  await page.mouse.wheel(0, 10)
  await page.mouse.wheel(0, 10)
  await expect(input).toHaveValue('1111')
```